### DISCUSSION TITLE
Models Supported - Do you have a different model than this list?

### DISCUSSION THREAD

**Discussion by D-Jeffrey** [crowbarz/ha-pioneer_async#37] [General]
These models I note either have custom configuration or people have noted that they are using that model and we can assume it works.  Are there any other models that people have used this on?

# Confirmed Working (as at 2026-03-13)
| SC-xx     | SC-1xxx   | SC-2xxx   | SC-LXxx     | VSX-xx    | VSX-xxx   | VSX-1xxx  | VSX-2xxx
| ---       | ---       | ---       | ---       | ---       | ---       | ---       | ---
| SC-71     | SC-1223   | SC-2023   | SC-LX57   | VSX-45    |  VSX-528   | VSX-1020  | VSX-2020
| SC-75     | SC-1228   | SC-2024   | SC-LX73   |           |VSX-529 |  VSX-1025 *| SX-2021
| SC-77     | SC-1323   |           | SC-LX77   |           | VSX-822 + ! | VSX-1028 |
|           | SC-1328   |           | SC-LX78   |           | VSX-823 + ! |    VSX-1121 |
|           | SC-1523   |           | SC-LX87   |           | VSX-830    |  VSX-1123|
|           | SC-1528   |           |           |           |VSX-921 |  VSX-1128 |
|           |           |           |           |           | VSX-922   |           |
|           |           |           |           |          | VSX-923   |  VSX-1325    |
|           |           |           |           |           | VSX-924   |           |
|           |           |           |           |           | VSX-925 * |           |
|           |           |           |           |           | VSX-930   |           | 

### Caveats
\+ set settings `Query sources from AVR` as `Disable` 
! has tuner but cannot control
 \* does not support offical sleep mode - VSX-31, VSX-1025 and VSX-925 do not have a standby mode, but you can keep it awake by turned enabling MultiZone and set Zone 2 On.

# Compatible? - TDB
| VSX |  
| --- |
VSX-838 
VSX-1122

... Awaiting logs or a comment that Onkyo is a better choice?

## Pioneer AVRs released from 2016 onwards use the Onkyo API
# NOT

| VSX           |  VSX-LX      | SC-LX       |
|------------ |---------      |---------     |
|VSX-933 ^ | VSX‑LX102|   SC‑LX701
|VSX-935 ^|  VSX‑LX302|   SC-LX801
|                 | VSX‑LX503|   SC-LX901
|                 |  VSX-LX505 |    

Try using one of the [Onkyo](https://github.com/zarpli/onkyo) or [onkyo-eiscp](https://github.com/miracle2k/onkyo-eiscp) integrations instead.
^ confirmed to work with Onkyo HA Module

### What the “‑K” or "-S" means in Pioneer Model Numbers
Pioneer uses suffix letters to indicate finish/color:

| Suffix | Meaning|
|-- | --|
K | Black finish
S | Silver finish
(no suffix) | Region‑dependent, often defaults to black

### Model Series
Series | Target Consumer
--- | ---
VSX | Entry to Mid‑Range, traditional Class AB amplifiers - mainstream, affordable, good for most setups
SC | High‑End with Class D Amplification (D3 digital amplifiers - Pioneer’s proprietary design) 
SC‑LX | Flagship Audiophile Line build and components (also known as LX Series in Europe)

**Comment by D-Jeffrey** [comment]
Can you add a Label of 'Question'

**Comment by pantherale0** [comment]
FYI, a lot of the configurations that were added in aiopioneer 0.2.0 were taken from the Pioneer docs, I can't guarantee that they will 100% work.

It would be worth asking the release year of each AVR and firmware version too if known. That way it helps when creating a configuration for that AVR as it can be cross referenced to the Pioneer docs.

**Comment by Toets** [comment]
Please add my old VSX-2021. Config flow didn't work for me but yaml does.

**Comment by pantherale0** [comment]
@Toets can you provide your yaml config please?

**Comment by Toets** [comment]
media_player:
  - platform: pioneer_async
    name: AVR WK
    host: 10.100.15.22
    port: 8103

**Comment by pantherale0** [comment]
Thats interesting, those options are configurable via the config flow. What happens when you configure with the config flow?

You'll need to change the port from the default value.

**Comment by D-Jeffrey** [comment]
@11harveyj I find that [hifiengine](https://www.hifiengine.com/manual_library/pioneer.shtml) is good for looking up older models. VSX-2021 is 2011/2012.  Since @crowbarz added the prompts in the flow for the port number in the Add interface, it can be adjusted (only at that time) without having to add configuration.yaml.  So far, I have found a couple of variances that make the device work differently - volume step, no standby enabled/supported and no response to ?SVB.

Reading the VSX-1021 manual, I see that you can adjust the port number.
`Home menu-> System Setup -> Network Setup -> `**`Port Number Settings`**
And in the manual, it suggests 23.  It also indicates that you can have up to 5 port numbers set.

@Toets is that the case on your receiver?  Can you add multiple port numbers? and it adjustable there?

Maybe there are some 'Adjust your Pioneer for best experience' notes that should be in the readme.

**Comment by D-Jeffrey** [comment]
@11harveyj I created a list of the possible models: 56 in all which includes the year it was made [Notes](https://github.com/D-Jeffrey/pioneer_vsx/blob/master/DeveloperNote.md) & [JSON](https://github.com/D-Jeffrey/pioneer_vsx/blob/master/pioneermodel.json)

**Comment by pantherale0** [comment]
Looks good, thank you, just to note, my SC-LX57 supports network, network standby and pretty much all other commands too (it's what was used when I was building out all the commands for aiopioneer)

From my point of view all AVRs with a model number starting with SC-LX and newer than 2012 (maybe 2011) supports most features, I'm led to believe these were the "best of the best" 10~ yrs ago.

There's a good amount of docs for most AVRs starting SC too, the VSX AVRs are where things get tricky because not so much exists for those (such as yours for example).

I think it would be good to build a dev tool to send and capture as much protocol related information about an AVR as possible, mainly for these AVRs where we're not sure on whats supported and what isn't.

**Comment by D-Jeffrey** [comment]
I will have to see if there is a better repository out there that I can get a improved answer from.  Is there a way that we could push into to the 'analytics' to build up information on all the different models (properly) that choose to install this integration?  I wonder if there is a way, we could leverage this idea [processing](https://www.home-assistant.io/integrations/analytics/#data-storage--processing)?

I see in the data [[Analytics](https://analytics.home-assistant.io/)  I see there are: `390 | Pioneer | 221 (0.1%)` of the built in Pioneer model.  (I wonder how many people removed the default Pioneer and are trying other customs) like:
@mk3l @BIoob @lakee911 @CodingSpiderFox @leodirob @jtonk @tiagofreire-pt @user0x01 @realthk and so many more.)

**Reply by D-Jeffrey** [reply]
Currently there are only 2 users of the built in Pioneer module.

**Comment by Toets** [comment]
@D-Jeffrey: Yes, ports can be added on my VSX-2021 as you described. Ever since I owned this AVR I experienced issues controlling over the network. I added a port long time ago while investigating the issue. Using telnet I can simply see the AVR gets unresponsive on a particular port after a while. A power cycle fixes the issue. It is an understatement to say the VSX-2021 is buggy...

In the config flow I change the host IP address and port number. Pressing Submit it immediately reports Cannot connect to AVR and input fields change back to default. Tired of troubleshooting due to buggy AVRI fell back to yaml and am able to somewhat make this work. ha-pioneer_async is a huge improvement over the standard pioneer integration but still far from stable in my scenario.

**Comment by pantherale0** [comment]
@Toets there are some comments about the stability of the Pioneer LAN interface here https://github.com/crowbarz/aiopioneer/discussions/18

In essence, I think its due to the fact the AVR does not properly handle the termination of TCP network connections, with its limited memory they get to a point where they cannot handle any new connections as you have described. Do you use the Pioneer app at all or anything else connecting to the AVR? These issues seem to affect Pioneer AVRs regardless if you navigate to the web server on it, or use the serial connection.

**Comment by MahrWe** [comment]
I have the VSX-921, it's connected to my network via LAN. I cannot change the port, it's fixed to 8103. 
I can confirm the mentioned stability issues (using the onkyo ha-integration), this is why I moved from ha-pioneer/onkyo integration to this one. This has worked for me very well ever since.

**Reply by MahrWe** [reply]
I have the VSX-921k and it works well with the integration. So I am confused about the entry under "not compatible"

**Reply by The-M1k3y** [reply]
Same here. Using this integration for a few months now with a VSX-921. No issues so far.

**Comment by rarroyo6** [comment]
The VSX-823 and VSX-822 can be added to the list. I have one running, and can verify that most of it works fine with this integration.
The following does not work:
- The sources have to be added manually, they are not automatically detected.
- The volume must be stepped up or down, it can not be set directly.
- The AM/FM band select and frequency set does not work.

**Comment by DeckardCain2014** [comment]
Does anyone has the VSX-932 successfully setup? I can't connect with port 23 nor 8102

**Comment by crowbarz** [comment]
> Does anyone has the VSX-932 successfully setup? I can't connect with port 23 nor 8102

Try power cycling the AVR, especially if you have previously used the core pioneer integration.

Does the AVR respond to ping? Are you able to use the relevant Pioneer app to remotely control the AVR?

You can also try to manually connect to your AVR via `telnet` or `nc` to ensure that the AVR is responding on either port.

**Comment by jasongabler** [comment]
I have a VSX-LX503.  I can't even get the device to appear in the devices list after configuration and reboot. 

Here's my config:

```
media_player:
  - platform: pioneer_async
    name: Pioneer AVR
    host: 192.168.0.134
    port: 8102
    scan_interval: 60
    timeout: 5.0
    params:
      ignore_volume_check: true
```

Surely, part of the issue is that my receiver doesn't listen on that port:

```
% nmap 192.168.0.134 
Starting Nmap 7.94 ( https://nmap.org ) at 2023-08-12 17:39 PDT
Nmap scan report for 192.168.0.134
Host is up (0.0027s latency).
Not shown: 991 closed tcp ports (conn-refused)
PORT      STATE SERVICE
80/tcp    open  http
7000/tcp  open  afs3-fileserver
8008/tcp  open  http
8009/tcp  open  ajp13
8080/tcp  open  http-proxy
8443/tcp  open  https-alt
8888/tcp  open  sun-answerbook
9000/tcp  open  cslistener
10001/tcp open  scp-config
```

I tried all of those ports in the config. All of them gave something like or similar to:

```
2023-08-12 19:50:31.625 WARNING (MainThread) [homeassistant.components.media_player] Setup of media_player platform pioneer_async is taking over 10 seconds.
2023-08-12 19:50:41.633 ERROR (MainThread) [custom_components.pioneer_async.media_player] Could not open AVR connection: RuntimeError: Main Zone not found on AVR
2023-08-12 19:50:41.635 WARNING (MainThread) [homeassistant.components.media_player] Platform pioneer_async not ready yet: None; Retrying in background in 30 seconds
```

Any ideas? 

Thanks.

**Comment by pantherale0** [comment]
> I have a VSX-LX503. I can't even get the device to appear in the devices list after configuration and reboot.
> 
> Here's my config:
> 
> ```
> media_player:
>   - platform: pioneer_async
>     name: Pioneer AVR
>     host: 192.168.0.134
>     port: 8102
>     scan_interval: 60
>     timeout: 5.0
>     params:
>       ignore_volume_check: true
> ```
> 
> Surely, part of the issue is that my receiver doesn't listen on that port:
> 
> ```
> % nmap 192.168.0.134 
> Starting Nmap 7.94 ( https://nmap.org ) at 2023-08-12 17:39 PDT
> Nmap scan report for 192.168.0.134
> Host is up (0.0027s latency).
> Not shown: 991 closed tcp ports (conn-refused)
> PORT      STATE SERVICE
> 80/tcp    open  http
> 7000/tcp  open  afs3-fileserver
> 8008/tcp  open  http
> 8009/tcp  open  ajp13
> 8080/tcp  open  http-proxy
> 8443/tcp  open  https-alt
> 8888/tcp  open  sun-answerbook
> 9000/tcp  open  cslistener
> 10001/tcp open  scp-config
> ```
> 
> I tried all of those ports in the config. All of them gave something like or similar to:
> 
> ```
> 2023-08-12 19:50:31.625 WARNING (MainThread) [homeassistant.components.media_player] Setup of media_player platform pioneer_async is taking over 10 seconds.
> 2023-08-12 19:50:41.633 ERROR (MainThread) [custom_components.pioneer_async.media_player] Could not open AVR connection: RuntimeError: Main Zone not found on AVR
> 2023-08-12 19:50:41.635 WARNING (MainThread) [homeassistant.components.media_player] Platform pioneer_async not ready yet: None; Retrying in background in 30 seconds
> ```
> 
> Any ideas?
> 
> Thanks.

This is something for a separate issue, but you need to make sure you can use the iControlAV 5 app with your AVR model. It might be too new for this integration as it looks like it doesn't expose a serial connection via the LAN.

Make sure network standby is turned on and perhaps l auto standby is turned off.

If it's still not working, test with the onkyo integration and see if that works instead. Otherwise I'm all out of ideas.

**Comment by jasongabler** [comment]
Yeah, I established a long time ago that it doesn't work with the Onkyo
integration.  I know at some point that Onkyo purchased Pioneer, but it
took a while for Pioneer branded AVRs to start using the (far superior)
Onkyo protocols. I think the LX503 was one of the last to use the old
Pioneer communication.   FWIW, my receiver can be controlled via
https://github.com/miracle2k/onkyo-eiscp.

**Comment by D-Jeffrey** [comment]
@pantherale0, A couple things to note and try. 

`nmap` does not automatically scan all the ports, just well-known ports without parameters.

```
$ nmap 192.168.4.201
Starting Nmap 7.80 ( https://nmap.org ) at 2023-08-13 21:29 MDT
Nmap scan report for 192.168.4.201
Host is up (0.049s latency).
Not shown: 999 closed ports
PORT   STATE SERVICE
80/tcp open  http

Nmap done: 1 IP address (1 host up) scanned in 0.73 seconds
````

if you use the port number it will show 8102 as active.  
````
$ nmap 192.168.4.201 -p 1-9000
Starting Nmap 7.80 ( https://nmap.org ) at 2023-08-13 21:33 MDT
Nmap scan report for 192.168.4.201
Host is up (0.049s latency).
Not shown: 8998 closed ports
PORT     STATE SERVICE
80/tcp   open  http
8102/tcp open  kz-migr

Nmap done: 1 IP address (1 host up) scanned in 8.95 seconds
````

You can try this url and see if it gives a response. 
[http://192.168.0.134/StatusHandler.asp](http://192.168.0.134/StatusHandler.asp)  you can see some more of the different links you can use [My Pioneer Info](https://github.com/D-Jeffrey/Pioneer-info)

I would also check page 149 of [VSX-LX503 Manual](https://intl.pioneer-audiovisual.com/manuals/docs/28d8197e3064e03602937fe0afb9de235f4b60ab.pdf) and see if the port has been changed from the default 8102.  If you are using Control4, then turn off the app while you try and configure.  Some versions can not handle multiple apps connecting to it at the sametime.

I would take the configuration out of your `configuration.yaml` and just using the configuration option after you add the Pioneer AVR module to HA and add a device in the interface instead.  

Update on updated on what you find.

**Comment by BIoob** [comment]
I'm using it with a VSX-924

**Comment by jasongabler** [comment]
> @pantherale0, A couple things to note and try.
> 
> `nmap` does not automatically scan all the ports, just well-known ports without parameters.
> 
> ```

Thanks. I took your suggestion and the resulting list was longer. Unfortunately, the same result with none of the ports satisfying this HACS plugin:

```
% nmap 192.168.0.134 -p 1-65535      
Starting Nmap 7.94 ( https://nmap.org ) at 2023-08-20 13:11 PDT
Nmap scan report for 192.168.0.134
Host is up (0.0029s latency).
Not shown: 65517 closed tcp ports (conn-refused)
PORT      STATE SERVICE
80/tcp    open  http
4545/tcp  open  worldscores
7000/tcp  open  afs3-fileserver
8008/tcp  open  http
8009/tcp  open  ajp13
8080/tcp  open  http-proxy
8443/tcp  open  https-alt
8554/tcp  open  rtsp-alt
8888/tcp  open  sun-answerbook
9000/tcp  open  cslistener
9013/tcp  open  unknown
10001/tcp open  scp-config
10109/tcp open  unknown
10110/tcp open  nmea-0183
10111/tcp open  nmea-onenet
10113/tcp open  netiq-endpoint
10120/tcp open  unknown
60128/tcp open  unknown
```

> You can try this url and see if it gives a response. http://192.168.0.134/StatusHandler.asp you can see some more of the different links you can use [My Pioneer Info](https://github.com/D-Jeffrey/Pioneer-info)

This returned a 404. 

> I would also check page 149 of [VSX-LX503 Manual](https://intl.pioneer-audiovisual.com/manuals/docs/28d8197e3064e03602937fe0afb9de235f4b60ab.pdf) and see if the port has been changed from the default 8102. If you are using Control4, then turn off the app while you try and configure. Some versions can not handle multiple apps connecting to it at the sametime.

What exactly did you see on pg 149?  I've scoured this entire manual many times and there is absolutely no mention of network ports, numerically or otherwise -- albeit this page does say "using an Internet browser", which alludes to HTTP (80). And, HTTP does work for accessing the minimal network config UI for which you can see only one view in that manual page. This UI is nothing more than the most basic network setup reporting and TCP/IP config, as well as minimal C4 and Airplay config (neither of which I use.)

> 
> I would take the configuration out of your `configuration.yaml` and just using the configuration option after you add the Pioneer AVR module to HA and add a device in the interface instead.

Hmmm, I've never seen this "configuration option".  My HA is completely up to date: OS, HA itself, all add-ons, plugins and HACS plugins, all at the latest. Here's the plugin's tile in the HACS main screen:

<img width="502" alt="image" src="https://github.com/crowbarz/ha-pioneer_async/assets/1828132/407e22f8-286b-42d1-bcdf-6c733cdf262f">

If I click on the tile I see:

<img width="902" alt="image" src="https://github.com/crowbarz/ha-pioneer_async/assets/1828132/1cf44e6a-ffe8-430b-ae9f-a85f4baef283">


> 
> Update on updated on what you find.

Unfortunately, nothing helpful so far.

**Comment by crowbarz** [comment]
@jasongabler you mentioned earlier that your VSX-LX503 _is_ controllable using https://github.com/miracle2k/onkyo-eiscp ? Unfortunately this integration doesn't implement the protocol used by that repo - from a quick scan of [the commands](https://github.com/miracle2k/onkyo-eiscp/blob/master/eiscp-commands.yaml) it doesn't look much like the Pioneer protocol. Also, the HA integration complaining about the main zone not found would suggest that it can connect to the AVR but isn't responding as expected to the Pioneer commands that the integration is sending.

The VSX-LX503 manual does mention (on page 131) that it may be controllable via the iControlAV5 application from Pioneer, though that could just be a copy/paste artefact. As @pantherale0 mentioned, if it works with that app then it should work with this integration.

I see there is an RS-232 port on the back of a VSX-LX503, so a (very) long shot would be to connect something that exposes the serial port as a network connection (such as a terminal server) and point the integration at that device. You'd want to confirm which protocol is being used by the AVR before going down that route, however.

**Comment by pantherale0** [comment]
@jasongabler that AVR looks to be too new if I'm reading that port scan right.

The eISCP port is 60128, which yours appears to be listening on and it also looks like some of the Chromecast ports are listed there too.

FWIW - it's probably safe to assume that AVRs with eISCP / Chromecast built-in this integration won't work. Unless the serial port exposes the old pioneer protocol, but I wouldn't go out of my way to buy hardware to test as it probably exposes onkyo's serial protocol which is quite different.

https://github.com/zarpli/onkyo

**Comment by jasongabler** [comment]
@crowbarz  and @pantherale0  ,

Thanks for the replies and insights. I thought my receiver was speaking a protocol that was too old. Seems like I had that backwards.  I'm pretty familiar with the eISCP stuff, at least enough to have made a utility to normalize volumes between zones (python running on UN*X).  I suppose I could finally make a foray into hacking HA and contributing to this project... or wait for the rest of you to upgrade your receivers :D   For now I'll have to stick with my Harmony Hub as the intermediary.

Thanks.

**Comment by lhassell** [comment]
I can confirm working on VSX-1021.

**Comment by cyberdiamond** [comment]
SC-LX73 confirmed working (inc auto detection of sources but only 2 of the 3 available zones added?), thanks.

**Comment by MacPro-de** [comment]
My SC-2024 also seems to work fine (Port 8102). Also my two older VSX-2021 (Port 23). Thank you for the integration.

**Comment by bobbeims** [comment]
I just purchased a VSX-935 and tried to set up the integration, I'm getting the error below.

<img width="606" height="560" alt="image" src="https://github.com/user-attachments/assets/1bf8dbbe-d5c7-4c9f-b805-2e40728944a7" />

However, the Onkyo integration does stand up an entity that works (with limited features):

<img width="578" height="402" alt="image" src="https://github.com/user-attachments/assets/acd9f972-a113-43db-84e8-f65c24c24490" />

Do I need to remove the Onkyo instance so it doesn't collide with yours?

**Reply by crowbarz** [reply]
Unfortunately, if your AVR works with the Onkyo integration, then it won't work with this integration. As mentioned in the README: `Pioneer AVRs released from 2016 onwards use the Onkyo API, and will not work with this integration`. I will add a note to indicate that the AVR might also refuse to connect as in your case.

**Reply by D-Jeffrey** [reply]
The most complete list is at the top of this discussion, while the README points to https://github.com/crowbarz/ha-pioneer_async/issues/20

**Reply by crowbarz** [reply]
@D-Jeffrey the README has been updated now to point to this post

**Comment by daviddu26** [comment]
Hello. A big thanks to the community and @crowbarz for the work done on this project. I've recently set up an (admittedly fairly old) Pioneer VSX-528-K and was trying to connect it to Home Assistant through this integration. The integration detects the right model but unfortunately not much is working beyond this first step. I was wondering if somehow I could help to develop compatibility with this specific model. I am not a developper but absolutely willing to tinker and test and investigate however is needed if someone can guide me through the steps. Ideally I would love to get at least the surround mode selection working and of course some sort of power on/off. Let me know how I can help ?

**Reply by D-Jeffrey** [reply]
I'm going to take a leap and say a VSX-528 should work (questions is which integration).  If could be that you need to use the https://github.com/zarpli/onkyo instead since it appears that the 528 has DLNA services.   Let's check a few things first and discover the basics.  

If you put your Pioneer IP address in here 
http://`192.168.0.250`/StatusHandler.asp
and you get a 404 error then you should try the Onkyo.  

If you get something like the results of [github.com/D-Jeffrey/Pioneer-info](https://github.com/D-Jeffrey/Pioneer-info?tab=readme-ov-file#developer-notes-and-learnings) then keep going with this HA integration.  Whatever the results, please come back so the compatibility table can be updated.

**Reply by daviddu26** [reply]
I do get a error message if I try to connect to the http:receiverIP/StatusHandler.asp page.
The Onkyo integration is not working however. It's not picking up the receiver at all.
<img width="388" height="303" alt="Capture d’écran 2025-12-22 à 09 08 40" src="https://github.com/user-attachments/assets/ece377b4-3d5d-49ff-bcad-2330b11890be" />

This integration is the most promising lead as it it picking up the receiver and even detecting the right model automatically.
<img width="388" height="834" alt="Capture d’écran 2025-12-22 à 09 14 16" src="https://github.com/user-attachments/assets/1d39bcf2-dbe3-4489-a213-e3bcdd2e9598" />

However if I go through the initialization process (by adding at least one source). I get this:
<img width="1105" height="521" alt="Capture d’écran 2025-12-22 à 09 17 41" src="https://github.com/user-attachments/assets/39297031-382b-4708-9a18-c7e10a72a398" />

But then most of the entities are not functional:
<img width="1067" height="1104" alt="Capture d’écran 2025-12-22 à 09 20 09" src="https://github.com/user-attachments/assets/47e90a93-dfb3-4646-8370-ae469d074d5b" />

Is there anything that can be done ?

**Reply by crowbarz** [reply]
It looks like your AVR is being detected by this integration with main zone and zone 2, which is why the integration page shows three devices were added: one for the AVR itself, and also one for each zone. Many of those entities on the global AVR device page don't work unless you first power on the main zone. Your last screenshot is the device page for the global AVR - this page only contain the entities to change AVR wide settings. The zone specific entities are associated with the zone devices, including the `media_player` entitiy that control zone power, source selection and sound mode if supported.

The tuner entities also require a tuner source to be selected in one of the zones for them to become available. They are on the global AVR page because AFAIK there is only ever one tuner on an AVR, though on some models the tuner source is available on zones other than the main zone.

Note that not all AVRs support all features, so even if the entities activate your AVR may return an error if you try to change their values. Some AVRs do not even support setting volume level and/or toggling zone power, you will need to experiment with your AVR.

If you still can't get the global entities to work and you have verified (in the AVR manual) that the feature is available on your AVR model, then please open a separate issue in this repo and include the debug logs, which you can enable via the overflow menu on the integration page showing the list of devices. HA will prompt a download for the logs on disabling debug.

**Reply by daviddu26** [reply]
I've opened an issue documented with my findings. I'm hoping this can help bring better support for the VSX-528. Thanks !

**Reply by crowbarz** [reply]
In the OP you said:

> Ideally I would love to get at least the surround mode selection working and of course some sort of power on/off. Let me know how I can help ?

Do the functions you mentioned work after turning on the AVR, either via the integration using the appropriate device controls, or manually?

Virtually all of the commands that you listed in your issue are already directly supported by the integration. Input sources are configurable in the integration as well, and it will be up to you to discover which sources exist on your AVR. There is already a mapping of known input source IDs to source names when you click through the links in the basic options table entry for **Manually configured sources** in the README, this will help with your input discovery exercise. Mute and volume are also supported, and there is already a bug requesting improved support for network sources that present metadata of the item that is currently playing.

A debug log (as already mentioned above) during your attempts to use specific basic functions (like surround sound selection) would help a lot more here than an AI generated report about your AVR. My focus is fixing functions that **don't** work, rather than comprehensively adding support for an AVR platform that I don't have access to.

**Comment by gb0101010101** [comment]
Have Pioneer VSX-1122-K, which is not listed in supported AVRs, but most things seems to be working.

```
nmap home-studio-amp.lan -p 1-65535  
Starting Nmap 7.94SVN ( https://nmap.org ) at 2026-01-20 19:39 PST
Nmap scan report for home-studio-amp.lan (192.168.1.37)
Host is up (0.0019s latency).
Not shown: 65526 closed tcp ports (conn-refused)
PORT      STATE SERVICE
80/tcp    open  http
443/tcp   open  https
1024/tcp  open  kdm
1900/tcp  open  upnp
8080/tcp  open  http-proxy
8102/tcp  open  kz-migr
49152/tcp open  unknown
49153/tcp open  unknown
49154/tcp open  unknown

Nmap done: 1 IP address (1 host up) scanned in 9.96 seconds
```
Have it setup using mode `Zone 2`, with 5.1 speakers on `Main zone`, and 2.0 speakers on `Zone 2`. This creates entries in Pioneer AVR integration named:

- Studio Amp (75 entities)
- Studio Amp Main Zone (45 entities)
- Studio Amp Zone 2 (26 entities)

Some strange things are:

- When selecting a media player for action `media_player.select_sound_mode` I see two entries named  `Studio Amp Main Zone`. One is regular `Media Player` and the other is `Pioneer AVR`. I also have `Studio Amp DLNA` because I use DLNA for Xboxes. Do any of these integration conflict? There is only one `media_player.studio_amp_main_zone` entity reported in Developer tools states.
- When setting action `media_player.select_sound_mode` with target `Studio Amp Main Zone (Pioneer AVR)` there is a lag of about 60 seconds to update `sound_mode` state on `media_player.studio_amp_main_zone`, even though I hear the amp click instantly switching to it. If I do the same action using `Studio Amp Main Zone (Media Player)` it updates instantly.

Let me know if you want more info and if I should start a new issue for it.

**Reply by crowbarz** [reply]
I suspect that the first issue is due to both the device and the main entity being made available for target selection. If you show the YAML after selecting each option, the `target` key should be different. But the end result is the same. Target resolution is handled by HA, and the same function is called on the integration regardless of how you specify the target.

The target selection UI changed recently on HA to make the distinction between entities and devices more obvious. Which version of HA are you running?

The second issue is possibly due to your AVR not reporting the current sound mode after it is changed to HA, until the next 60s poll. The integration waits for the AVR to report the change before the change becomes visible in the UI, because the AVR may refuse to set an incompatible sound mode, or a cyclic sound mode has been used so the actual sound mode used by the AVR is not the one that was selected in the UI.

Can you log a separate bug on the second issue and include a [debug log](https://github.com/crowbarz/ha-pioneer_async?tab=readme-ov-file#enabling-debugging) covering the changing of the sound mode until the mode shows as changed on the AVR? The log should include the commands sent to and received from the AVR, which will help confirm whether this is indeed what is happening.

**Comment by modem-man-gmx** [comment]
Anybody knows, why my Pioneer VSX S520 is not found by the integration? I had it with the legacy integration. But somewhere between 2023 and 2025 an update of the legacy integration "degraded" the entities to kill all but the "power on/off". And perhaps supporting chromecast now (but may be this is entirely independently). So I thought giving the HACS a try would be cool.
 With powered on UKW radio, I started configuration (entered the DNS name my PiHole DNS is resolving for it), but even after hours, the VSX has not been found. Different Port?
Any hint?
Or, btw, a guide to go back to legal integration?

**Reply by D-Jeffrey** [reply]
I suggest that you look at the https://www.home-assistant.io/integrations/onkyo, since that is a newer Pioneer model.

**Reply by crowbarz** [reply]
I'm surprised that your AVR worked with the legacy integration at all. That integration just connects to an IP address and port and blindly sends hardcoded commands, with no error checking of the response other than the connection being refused. So I guess some basic Onkyo commands are/were similar to the Pioneer equivalents. I don't have an Onkyo receiver so have not looked at the Onkyo command protocol at all.

This integration validates that the device is responding like a Pioneer AVR device when adding an instance, so it will not allow you to add a device that doesn't respond to Pioneer commands.

**Reply by modem-man-gmx** [reply]
> I suggest that you look at the https://www.home-assistant.io/integrations/onkyo, since that is a newer Pioneer model.

THX! I installed the standard onkyo integration. Auto-Search for Onkyo did not work, but manually entering the network name did and the device got identified immediately by the correct type name. The control elements are now quite comfortable (On/Off, but also input source, sound mode, loudness, and a bunch of speaker adjustments I could use with own slider cards).

Merci!

**Reply by modem-man-gmx** [reply]
> I'm surprised that your AVR worked with the legacy integration at all. That integration just connects to an IP address and port and blindly sends hardcoded commands, with no error checking of the response other than the connection being refused. So I guess some basic Onkyo commands are/were similar to the Pioneer equivalents.

Yes, I can confirm this. At least the power on/off seems to be the same. And perhaps the volume control? I don't remember exactly, if volume was working or not.
 My best guess today is: I purchased a 2013 made unit with firmware which still has been "more Pioneerish". Then some time after 2023, I loaded the one and only "new" firmware, which is Onkyo in the meantime and this has been the day, I lost all the other Pioneer entities from HA. Onkyo/Pioneer fusion seems to happened around 2015, VSX-S520 got introduced in 2010. Could fit.
At least I remember doing a firmware some years before, hoping my 3D-DVD HDMI video pass-through problems will vanish. They didn't, but this is another story.

**Reply by modem-man-gmx** [reply]
Ugh, total recall!
 I lost the original VSX 520-S from 2013 in a thunderstorm. The recent unit I am talking about is already the slim line VSX **S**520-S, purchased in 2019. This unit **is** in fact a Onkyo TX-L50 plus steroids (MCACC) and just a Pioneer brand sticking on it.
 So, no, whatever the firmware update did, it did not made a formerly Pioneer becoming an Onkyo, because it already got shipped as Onkyo from 1st day on.
